lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -278,6 +280,89 @@ public Task transfer(String taskId, String destinationWorkbasketKey)
         }
     }
 
+    @Override
+    public BulkOperationResults<String, TaskanaException> transferBulk(String destinationWorkbasketKey,
+        List<String> taskIds) throws NotAuthorizedException, InvalidArgumentException, WorkbasketNotFoundException {
+        try {
+            taskanaEngineImpl.openConnection();
+            LOGGER.debug("entry to transferBulk(targetWbKey = {}, taskIds = {})", destinationWorkbasketKey, taskIds);
+            // Check pre-conditions with trowing Exceptions
+            if (destinationWorkbasketKey == null || taskIds == null) {
+                throw new InvalidArgumentException(
+                    "DestinationWorkbasketKey or TaskIds can´t be used as NULL-Parameter.");
+            }
+            Workbasket destinationWorkbasket = workbasketService.getWorkbasketByKey(destinationWorkbasketKey);
+
+            BulkOperationResults<String, TaskanaException> bulkLog = new BulkOperationResults<>();
+            // check tasks exist and Ids valid - log and remove
+            List<TaskSummary> taskSummaries = this.createTaskQuery().idIn(taskIds.toArray(new String[0])).list();
+            Iterator<String> taskIdIterator = taskIds.iterator();
+            while (taskIdIterator.hasNext()) {
+                String currentTaskId = taskIdIterator.next();
+                if (currentTaskId == null || currentTaskId.equals("")) {
+                    bulkLog.addError("",
+                        new InvalidArgumentException("IDs with EMPTY or NULL value are not allowed."));
+                    taskIdIterator.remove();
+                } else if (!taskSummaries.stream()
+                    .filter(taskSummary -> currentTaskId.equals(taskSummary.getTaskId()))
+                    .findFirst()
+                    .isPresent()) {
+                    bulkLog.addError(currentTaskId, new TaskNotFoundException(currentTaskId));
+                    taskIdIterator.remove();
+                }
+            }
+
+            // check source WB (read)+transfer
+            List<String> workbasketKeys = new ArrayList<>();
+            taskSummaries.stream().forEach(t -> workbasketKeys.add(t.getWorkbasketSummary().getKey()));
+            List<WorkbasketSummary> sourceWorkbaskets = workbasketService.createWorkbasketQuery()
+                .callerHasPermission(WorkbasketAuthorization.TRANSFER)
+                .keyIn(workbasketKeys.toArray(new String[0]))
+                .list();
+            taskIdIterator = taskIds.iterator();
+            while (taskIdIterator.hasNext()) {
+                String currentTaskId = taskIdIterator.next();
+                TaskSummary taskSummary = taskSummaries.stream()
+                    .filter(t -> currentTaskId.equals(t.getTaskId()))
+                    .findFirst()
+                    .get();
+                if (!sourceWorkbaskets.stream()
```

Comment:
findFirst() returns an Optional. .get() should only be called if you are sure that the optional is not empty. Otherwise you may want to call .orElse(null) rather than .get()

Reply:
done